packages/jitar-caching/src/building/SegmentReader.ts

```diff
@@ -0,0 +1,158 @@
+
+import { ReflectionFunction, ReflectionModule, Reflector } from 'jitar-reflection';
+import { FileManager } from 'jitar-runtime';
+
+import InvalidSegmentFilename from './errors/InvalidSegmentFilename.js';
+import MissingModuleExport from './errors/MissingModuleExport.js';
+import SegmentFileNotLoaded from './errors/SegmentFileNotLoaded.js';
+import SegmentModuleNotLoaded from './errors/SegmentModuleNotLoaded.js';
+
+import Segment from './models/Segment.js';
+import SegmentImplementation from './models/SegmentImplementation.js';
+import SegmentModule from './models/SegmentModule';
+import SegmentProcedure from './models/SegmentProcedure.js';
+
+import SegmentFile from './types/SegmentFile.js';
+import SegmentImports from './types/SegmentImports.js';
+
+const SEGMENT_FILE_EXTENSION = '.segment.json';
+const DEFAULT_ACCESS_LEVEL = 'private';
+const DEFAULT_VERSION_NUMBER = '0.0.0';
+
+const reflector = new Reflector();
+
+export default class SegmentReader
+{
+    #fileManager: FileManager;
+
+    constructor(fileManager: FileManager)
+    {
+        this.#fileManager = fileManager;
+    }
+
+    async read(filename: string): Promise<Segment>
+    {
+        const name = this.#extractSegmentName(filename);
+        const definition = await this.#loadSegmentDefinition(filename);
+        const modules = await this.#createSegmentModules(definition);
+
+        return new Segment(name, modules);
+    }
+
+    #extractSegmentName(filename: string): string
+    {
+        const file = filename.split('/').pop();
+
+        if (file === undefined)
+        {
+            throw new InvalidSegmentFilename(filename);
+        }
+
+        return file.replace(SEGMENT_FILE_EXTENSION, '');
+    }
+
+    async #loadSegmentDefinition(filename: string): Promise<SegmentFile>
+    {
+        try
+        {
+            const content = await this.#fileManager.getContent(filename);
+
+            return JSON.parse(content.toString()) as SegmentFile;
+        }
+        catch (error: unknown)
+        {
+            throw new SegmentFileNotLoaded(filename);
```

Comment:
Should we do something with the actual error to make it clear what went wrong before throwing a customized error?